WalletWasabi.Tests/UnitTests/Blockchain/TransactionBuilding/BranchAndBoundTests.cs

```diff
@@ -15,106 +16,195 @@ public class BranchAndBoundTests
 	[Fact]
 	public void MoreSelection_NoInputCosts()
 	{
+		using CancellationTokenSource cts = new(TimeSpan.FromSeconds(30));
+
 		long[] inputValues = new long[] { 11, 7, 5, 3, 2 }; // Sum is 28.
 		long[] inputCosts = new long[] { 0, 0, 0, 0, 0 }; // No input costs. Idealized.
 		long target = 27; // Target that we cannot get as a sum of input values.
 
 		BranchAndBound algorithm = new();
 		MoreSelectionStrategy strategy = new(target, inputValues, inputCosts);
-		bool wasSuccessful = algorithm.TryGetMatch(strategy, out List<long>? selectedCoins);
+		_ = algorithm.TryGetMatch(strategy, out _, cts.Token);
```

Comment:
We don't really know if a solution is best one because even if we have an exact match, we don't know if input costs cannot be lower. For this reason, tests were written in a way that it confused what the code does.

Obviously, it would be great to be able to say "this is the best solution we can have" but in reality, it may not be that important. I guess we will see if users are happy with the feature as is or if they are disappointed that it can be better.